src/components/structures/TimelinePanel.tsx

```diff
@@ -1394,6 +1398,72 @@ class TimelinePanel extends React.Component<IProps, IState> {
         return null;
     }
 
+    private createFormattedText = (events: MatrixEvent[]) => {
+        let content = "";
+        for (let i = 0; i < events.length; i++) {
+            const mxEv = events[i];
+            if (!mxEv || !haveTileForEvent(mxEv)) continue;
+            const senderDisplayName = mxEv.sender && mxEv.sender.name ? mxEv.sender.name : mxEv.getSender();
+            let text = "";
+            if (isReply(mxEv)) text = senderDisplayName + ": " + textForReplyEvent(mxEv.getContent());
+            else text = textForEvent(mxEv);
+            content += text && `${new Date(mxEv.getTs()).toLocaleString()} - ${text}\n`;
+        }
+        return content;
+    };
+
+    private getClosestEvent = (el: HTMLElement, fromTop: boolean): string => {
+        const messageList = document.querySelector("ol.mx_RoomView_MessageList");
+        let requiredElement: Element;
+        // if the selected element belongs to a date separator, assign its neighbouring element as the required element
+        if (el.parentElement.classList.contains("mx_DateSeparator")) {
+            while (el.parentElement != messageList) el = el.parentElement;
```

Comment:
This will not work since we are trying to find the closest event tile rather than message list. Also, the direction also matters (top or bottom). So, `closest` won't do the job.

Reply:
Not quite sure why this wouldn't work? Probably the selector is wrong, so update it with the correct one to query the correct element.

But in here you're looping and querying the `parentElement` on every iteration, which is what `closest` does? the `fromTop` logic is dealt with below if I read the code correctly?

Reply:
Yup, the selector has to be modified. Thought you were suggesting this as a replacement for the below code as well, sorry about that.